client.go

```diff
@@ -1,58 +1,90 @@
 package tcprouter
 
 import (
+	"context"
 	"fmt"
 	"io"
 	"net"
 
+	"github.com/libp2p/go-yamux"
 	"github.com/rs/zerolog/log"
 )
 
 type Client struct {
-	// connection to the tcp router server
-	RemoteConn net.Conn
-	// connection to the local application
-	LocalConn net.Conn
-
+	localAddr  string
+	remoteAddr string
 	// secret used to identify the connection in the tcp router server
 	secret []byte
+
+	// connection to the tcp router server
+	remoteSession *yamux.Session
 }
 
 // NewClient creates a new TCP router client
-func NewClient(secret string) *Client {
+func NewClient(secret string, local, remote string) *Client {
 	return &Client{
-		secret: []byte(secret),
+		localAddr:  local,
+		remoteAddr: remote,
+		secret:     []byte(secret),
 	}
 }
 
-func (c *Client) ConnectRemote(addr string) error {
+func (c Client) Start(ctx context.Context) error {
+	if err := c.connectRemote(c.remoteAddr); err != nil {
+		return fmt.Errorf("failed to connect to TCP router server: %w", err)
+	}
+
+	log.Info().Msg("start hanshake")
+	if err := c.handshake(); err != nil {
+		return fmt.Errorf("failed to hanshake with TCP router server: %w", err)
+	}
+	log.Info().Msg("hanshake done")
+
+	return c.listen(ctx)
+}
+
+func (c *Client) connectRemote(addr string) error {
 	if len(c.secret) == 0 {
 		return fmt.Errorf("no secret configured")
 	}
 
-	conn, err := net.Dial("tcp", addr)
+	tcpAddr, err := net.ResolveTCPAddr("tcp", addr)
```

Comment:
this returns only 1 addr even if we have 3 tcprouters on specific domain no?  also requires a port which we should ignore if not specified to be 18000

Reply:
a client always connect to a single remote router. If you need to connect to multiple router then you instantiate multiple client. Check the client main function to see how this works.
What's up with the port 18000 ?